articles/model-anatomy.adoc

```diff
@@ -0,0 +1,644 @@
+  title: String = "ESA"
+  budget: Integer = 23
+}
+
+class jaxa as "40:BusinessPartner" {
+  title: String = "JAXA"
+  budget: Integer = 42
+}
+
+pr ..> esa: externalParties [0]
+pr ..> jaxa: externalParties [1]
+
+pr .> coop: mainResponsible
+
+class pr2 as "90:Project" {
+  description: String = "Grow corn"
+}
+
+coop <. pr2: mainResponsible
+----
+
+A node's classifier describes the valid references.
+This description includes the references’ _name_, _key_, whether it is _optional_, whether we allow _multiple_ targets in the reference, and its _type_.
+All of them mean the same as for <<containments, containments>>, respectively.
+For multiple references, we keep track which target is listed at which position.
+The same target can appear several times in a multiple reference.
+This can help, for example, to describe steps: `steps: Task[1..*] = {generate, saveAll, compile, saveAll}` refers to Task `saveAll` twice.
+
+LionWeb does not support bi-directional references.
+They are very hard to maintain, and also hard to process (i.e. if we want to traverse the whole model).
+
+==== ResolveInfo
+For references, we store the target node's id, and additionally a `resolveInfo`.
+ResolveInfo is a string that somehow describes the target node.
+Typical candidate would be the target's node `name` (if the target node has such a property).
+To support this typical default behavior, LionWeb has a built-in interface `INamed` with one property `name: String` that might be implemented by any classifier.
+However, it's up to the application to select and set the appropriate resolveInfo.
+
+This serves two purposes:
+
+* In the space project example above, assume some app that displays the list of external parties for the project as `ESA: 23, JAXA: 42`.
+Now someone thinks there's no project left with ESA, and deletes node 87 (the ESA one).
+If the `externalParties [0]` reference only knew the target node id, the best possible display would be `!unknown target 87!, JAXA: 42` -- not very helpful to the user.
+With resolveInfo, we can display `!unknown target ESA!, JAXA: 42`.
+Now the user has a much better chance of tracking down the error.
+
+* We need to know the target node id in order to store it in a reference.
+But in parser-based systems, after parsing we know only some kind of identifier of a reference target.
+We might want to store this unresolved _abstract syntax tree_ (AST).
+A later linking step hopefully finds the actual target and fills in the proper target node id.
+
+Target node id, resolveInfo, or both must be set for each reference -- otherwise the reference is pointless.
+
+==== Unresolved references
+Assume a node `T` that happens to be the target of a reference `myRef` from node `A`.
+When we delete `T`, the reference stays untouched: `A` still contains a reference `myRef` with target node id `T` (and maybe some resolveInfo), but the target cannot be resolved.
+LionWeb allows references with unresolvable target.
```

Comment:
It's up to the consumer of the model to do something with/about that target, though.